File: js/sumOfLettersEvenOrOdd.js
// Problem# 271
// Create a function that takes a string and returns true if the sum of the position of every letter in the
// alphabet is even and false if the sum is odd.
// https://edabit.com/challenge/RMJwgK3MzAb5DQoFY

const isAlpha = (str) => {
  const sum = [...str.toLowerCase().replace(/[^a-z]/g, "")].reduce(
    (acc, curr) => acc + (curr.charCodeAt() - 96),
    0
  );

  return sum % 2 === 0;
};

module.exports = isAlpha;

File: python/sum_of_letters_even_or_odd.py
"""
Is the Sum of Letters Even or Odd?

Create a function that takes a string and returns True if the sum of the position of every letter in the alphabet is even and False if the sum is odd.

https://edabit.com/challenge/Mm8SK7DCvzissCF2s
"""


def is_alpha_sum_even(word: str) -> bool:
    alpha_sum = sum(ord(letter) for letter in word if letter.isalpha())
    return alpha_sum % 2 == 0
